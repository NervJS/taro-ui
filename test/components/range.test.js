import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtRange from '../../.temp/components/range/index'

describe('AtRange Snap', () => {
  it('render initial AtRange', () => {
    const componet = renderToString(<AtRange />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtRange -- props className', () => {
    const componet = renderToString(<AtRange className='test' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtRange -- props customStyle', () => {
    const componet = renderToString(<AtRange customStyle='color:red;' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtRange -- props sliderStyle', () => {
    const componet = renderToString(<AtRange sliderStyle='color:red;' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtRange -- props railStyle', () => {
    const componet = renderToString(<AtRange railStyle='color:red;' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtRange -- props trackStyle', () => {
    const componet = renderToString(<AtRange railStyle='color:red;' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtRange -- props value', () => {
    const componet = renderToString(<AtRange value={[0, 30]} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtRange -- props min', () => {
    const componet = renderToString(<AtRange min={10} value={[10, 30]} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtRange -- props max', () => {
    const componet = renderToString(<AtRange max={50} value={[10, 30]} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtRange -- props disabled', () => {
    const componet = renderToString(<AtRange disabled />)
    expect(componet).toMatchSnapshot()
  })
})
