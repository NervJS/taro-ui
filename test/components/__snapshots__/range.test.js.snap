// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtRange Snap render AtRange -- props className 1`] = `"<div class=\\"at-range test\\"><div style=\\"height:;\\" class=\\"at-range__container\\"><div class=\\"at-range__rail\\"></div><div style=\\"left:0%;width:0%;\\" class=\\"at-range__track\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div></div></div>"`;

exports[`AtRange Snap render AtRange -- props customStyle 1`] = `"<div style=\\"color:red;\\" class=\\"at-range\\"><div style=\\"height:;\\" class=\\"at-range__container\\"><div class=\\"at-range__rail\\"></div><div style=\\"left:0%;width:0%;\\" class=\\"at-range__track\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div></div></div>"`;

exports[`AtRange Snap render AtRange -- props disabled 1`] = `"<div class=\\"at-range at-range--disabled\\"><div style=\\"height:;\\" class=\\"at-range__container\\"><div class=\\"at-range__rail\\"></div><div style=\\"left:0%;width:0%;\\" class=\\"at-range__track\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div></div></div>"`;

exports[`AtRange Snap render AtRange -- props max 1`] = `"<div class=\\"at-range\\"><div style=\\"height:;\\" class=\\"at-range__container\\"><div class=\\"at-range__rail\\"></div><div style=\\"left:0%;width:0%;\\" class=\\"at-range__track\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div></div></div>"`;

exports[`AtRange Snap render AtRange -- props min 1`] = `"<div class=\\"at-range\\"><div style=\\"height:;\\" class=\\"at-range__container\\"><div class=\\"at-range__rail\\"></div><div style=\\"left:0%;width:0%;\\" class=\\"at-range__track\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div></div></div>"`;

exports[`AtRange Snap render AtRange -- props railStyle 1`] = `"<div class=\\"at-range\\"><div style=\\"height:;\\" class=\\"at-range__container\\"><div style=\\"color:red;\\" class=\\"at-range__rail\\"></div><div style=\\"left:0%;width:0%;\\" class=\\"at-range__track\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div></div></div>"`;

exports[`AtRange Snap render AtRange -- props sliderStyle 1`] = `"<div class=\\"at-range\\"><div style=\\"height:;\\" class=\\"at-range__container\\"><div class=\\"at-range__rail\\"></div><div style=\\"left:0%;width:0%;\\" class=\\"at-range__track\\"></div><div style=\\"width:;height:;margin-left:;left:0%;color:red;\\" class=\\"at-range__slider\\"></div><div style=\\"width:;height:;margin-left:;left:0%;color:red;\\" class=\\"at-range__slider\\"></div></div></div>"`;

exports[`AtRange Snap render AtRange -- props trackStyle 1`] = `"<div class=\\"at-range\\"><div style=\\"height:;\\" class=\\"at-range__container\\"><div style=\\"color:red;\\" class=\\"at-range__rail\\"></div><div style=\\"left:0%;width:0%;\\" class=\\"at-range__track\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div></div></div>"`;

exports[`AtRange Snap render AtRange -- props value 1`] = `"<div class=\\"at-range\\"><div style=\\"height:;\\" class=\\"at-range__container\\"><div class=\\"at-range__rail\\"></div><div style=\\"left:0%;width:0%;\\" class=\\"at-range__track\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div></div></div>"`;

exports[`AtRange Snap render initial AtRange 1`] = `"<div class=\\"at-range\\"><div style=\\"height:;\\" class=\\"at-range__container\\"><div class=\\"at-range__rail\\"></div><div style=\\"left:0%;width:0%;\\" class=\\"at-range__track\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div><div style=\\"width:;height:;margin-left:;left:0%;\\" class=\\"at-range__slider\\"></div></div></div>"`;
